crates/wasm-encoder/src/core/imports.rs

```diff
@@ -1,16 +1,26 @@
 use crate::{
-    CORE_FUNCTION_SORT, CORE_GLOBAL_SORT, CORE_MEMORY_SORT, CORE_TABLE_SORT, CORE_TAG_SORT, Encode,
-    GlobalType, MemoryType, Section, SectionId, TableType, TagType, encode_section,
+    CORE_FUNCTION_EXACT_SORT, CORE_FUNCTION_SORT, CORE_GLOBAL_SORT, CORE_MEMORY_SORT,
+    CORE_TABLE_SORT, CORE_TAG_SORT, Encode, GlobalType, MemoryType, Section, SectionId, TableType,
+    TagType, encode_section,
 };
 use alloc::vec::Vec;
 
+/// The imported function type.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct FunctionHeapType {
+    /// A function type index.
+    pub ty: u32,
+    /// Exact type.
+    pub exact: bool,
+}
+
 /// The type of an entity.
 #[derive(Debug, Clone, Copy, PartialEq, Eq)]
 pub enum EntityType {
     /// A function type.
     ///
     /// The value is an index into the types section.
-    Function(u32),
+    Function(FunctionHeapType),
```

Comment:
To bikeshed this a bit, this might integrate better with existing `match`es and such with a similar pattern as to wasmparser where a new `FunctionExact` variant is added to the `enum` rather than using `exact: bool` on preexisting variants